coordinator/src/escrow_coordinator/mod.rs

```diff
@@ -56,16 +56,16 @@ impl EscrowCoordinator {
                                     EscrowCoordinator::parse_contract(&rumor.content)
                                 {
                                     debug!("Received contract: {}", &contract.trade_description);
-                                    if self.pending_contracts.contains_key(&contract_hash) {
+                                    if let std::collections::hash_map::Entry::Vacant(e) = self.pending_contracts.entry(contract_hash) {
```

Comment:
usage of `contains_key` followed by `insert` on a `HashMap`
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_entry
`#[warn(clippy::map_entry)]` on by defaultclippy[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2Fworkspaces%2Fcashu-escrow-kit%2Fcoordinator%2Fsrc%2Fescrow_coordinator%2Fmod.rs)
mod.rs(67, 37): try: `if let std::collections::hash_map::Entry::Vacant(e) = self.pending_contracts.entry(contract_hash) {
e.insert(contract);
} else {
self.pending_contracts.remove(&contract_hash);
let _ = self
.begin_trade(&contract_hash, &contract)
.await
.inspect_err(|e| {
error!("Got error while beginning a trade: {}", e);
});